internal/service/apikeyprojectassignment/resource.go

```diff
@@ -0,0 +1,176 @@
+package apikeyprojectassignment
+
+import (
+	"context"
+	"errors"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/validate"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+)
+
+const resourceName = "api_key_project_assignment"
+
+var (
+	_ resource.ResourceWithConfigure   = &rs{}
+	_ resource.ResourceWithImportState = &rs{}
+)
+
+func Resource() resource.Resource {
+	return &rs{
+		RSCommon: config.RSCommon{
+			ResourceName: resourceName,
+		},
+	}
+}
+
+type rs struct {
+	config.RSCommon
+}
+
+func (r *rs) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = ResourceSchema(ctx)
+	conversion.UpdateSchemaDescription(&resp.Schema)
+}
+
+func (r *rs) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var tfModel TFModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &tfModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	assignmentReq, diags := NewAtlasCreateReq(ctx, &tfModel)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
```

Comment:
```suggestion
	resp.Diagnostics.Append(diags...)
	if diags.HasError() {
		return
	}
```

in this way we don't miss warnings.   if no errors or warnings, Append is a no-op.

same in other parts of the code

Reply:
fixed, thanks